### PR TITLE
Deprecate Copy Behavior to support Quantities 16.0

### LINKED ISSUE #1529: Deprecation in pq.Quantity of the copy argument
Spotted today on CIs in MNE-Python: https://github.com/mne-tools/mne-python/actions/runs/10577451925/job/29305474492?pr=12815#step:17:5232

```
 neo/io/proxyobjects.py:296: in load
    anasig = AnalogSignal(
neo/core/analogsignal.py:202: in __new__
    obj = pq.Quantity(signal, units=units, dtype=dtype, copy=copy).view(cls)
quantities/quantity.py:120: in __new__
    warnings.warn(("The 'copy' argument in Quantity is deprecated and will be removed in the future. "
E   quantities.QuantitiesDeprecationWarning: The 'copy' argument in Quantity is deprecated and will be removed in the future. The argument has no effect since quantities-0.16.0 (to aid numpy-2.0 support).
```

### LINKED ISSUE #1534: Fix tests for quantities 0.16.0
core-tests failing with new version of quantities. We will need to fix.

### PR DESCRIPTION
**Pull request by zm711**
I want to do this in a couple parts. First remove copy. Then ensure tests are passing (currently I've commented out the tests, but my plan is to delete the copy=True tests). Just so people can monitor progress I'll just have this open as a PR.

Fixes #1534 
Fixes #1529

### CODE REVIEW

**Comment by zm711** [comment]
Things I found for us to discus:

1) we can't change the units or dtype during construction since we only get a view of the array. I deleted all tests changing units/dtype
2) time_shift required a different strategy so I implemented it
3) deleted all tests checking for copy vs view

4) I need help with the pickle tests. Reading from pickle just returns a None instead of the appropriate array. Someone who knows pickle better should check this for me. Currently they are commented out.

**Comment by zm711** [comment]
IO tests failing due to a CI issue right now.... We can check that later.

**Comment by h-mayorquin** [comment]
Try flushing the conda cashes just to see if the dependency issue goes away.

**Comment by zm711** [comment]
So now we have an issue in nix that I have propagated the fix for quantities [here](https://github.com/python-quantities/python-quantities/pull/242). 

I just have pickleio that is broken and I don't know why. @samuelgarcia or @apdavison can either of you walk me through the pickling story a bit more.

**Comment by zm711** [comment]
@apdavison, if you have the brain power to look over this. We are running into one stumbling block with pickleio not working. Sam and I looked at it and we aren't quite sure why. The nix test failure needs to be fixed with my PR at quantities level.

File: neo/core/irregularlysampledsignal.py [hunk 1]
```diff
@@ -544,6 +550,9 @@ def time_shift(self, t_shift):
             (the original :class:`IrregularlySampledSignal` is not modified).
         """
         new_sig = deepcopy(self)
+        # As of numpy 2.0/quantities 0.16 we need to copy the array itself
```
Comment (zm711) [inline neo/core/irregularlysampledsignal.py:553]: To be discussed.

File: neo/test/coretest/test_analogsignal.py [hunk 2]
```diff
@@ -1738,8 +1716,11 @@ def test_concatenate_multi_signal(self):
 
 
 class TestAnalogSignalFunctions(unittest.TestCase):
+
+    ## someone with more pickle knowledge needs to work on this
```
Comment (zm711) [inline neo/test/coretest/test_analogsignal.py:1720]: @apdavison

File: neo/test/coretest/test_irregularysampledsignal.py [hunk 3]
```diff
@@ -1033,6 +1033,8 @@ def test_concatenate_array_annotations(self):
 
 
 class TestAnalogSignalFunctions(unittest.TestCase):
+    # pickle help needed
```
Comment (zm711) [inline neo/test/coretest/test_irregularysampledsignal.py:1036]: @apdavison

**Comment by samuelgarcia** [comment]
@zm711 : I fixes the pickleio tsuff it was on the copy=True was still here in the `__reduce__` (used by pickle)

**Comment by zm711** [comment]
Perfect thanks for the pickleio. I'll uncomment the core pickle and then we have the nix fix that needs to happen in quantities.

**Comment by zm711** [comment]
@apdavison, I convinced Sam to help with the pickle a bit more. So last thing we need for tests to pass is for a release of quantities with https://github.com/python-quantities/python-quantities/pull/242 merged in. Tests have passed there. Then you can do the final review for this!

**Comment by apdavison** [comment]
https://pypi.org/project/quantities/0.16.1/

File: pyproject.toml [hunk 4]
```diff
@@ -23,8 +23,8 @@ classifiers = [
 
 dependencies = [
     "packaging",
-    "numpy>=1.22.4,<2.0.0",
```
Comment (apdavison) [inline pyproject.toml:26]: The dependency versions need to be fixed ("numpy>=1.22.4", "quantities>=0.16.1").

**Review by apdavison — APPROVED** [review]
The tests are now passing. Other than the fixes to version numbers in pyproject.toml, this looks fine.

File: pyproject.toml (hunk 4 above)
Comment (apdavison) [reply pyproject.toml:26]: it was more the lower limit I was talking about, which was going back to 1.19 when I reviewed this.

**Pull request merged**